### PR TITLE
Imports & Navbar visual changes

### PR DESCRIPTION
**Pull request by Ilverism**
09/11/24:
- Allows Environment Variable for SendEmail baseURL
    - Will default to hard-coded `"https://ngafid.org"` if unset
    - Should add `export NGAFID_BASE_URL="https://ngafid.org"` to `init_env.sh`
---
- Reworked how the Imports rows appear
- Modified the Uploads panel on the home page to reflect the new badges on the Imports page
- Added new icons to the Navbar
- Added a background highlight when hovering the Navbar
- Fixed the Navbar text not reflecting what the active page was for some pages

image
image

image
image

image
image

### CODE REVIEW

**Comment by jkarns275** [comment]
I like this

File: package.json [hunk 1]
```diff
@@ -20,17 +20,18 @@
         "babel-loader": "^8.2.2",
         "bootstrap": "^4.6.0",
         "cache-loader": "^1.0.3",
-        "html-loader": "^0.5.5",
-        "html-webpack-plugin": "^3.2.0",
+        "html-loader": "^5.0.0",
```
Comment (jkarns275) [inline package.json:23]: Reminds me that we need to keep all of our dependencies up to date... locking versions seems like a mistake?

File: src/main/java/org/ngafid/WebServer.java [hunk 2]
```diff
@@ -217,6 +217,7 @@ public static void main(String[] args) {
         Spark.post("/protected/monthly_event_counts", new PostMonthlyEventCounts(gson));
         Spark.get("/protected/severities", new GetSeverities(gson));
         Spark.post("/protected/severities", new PostSeverities(gson));
+        Spark.post("/protected/all_severities", new PostAllSeverities(gson));
```
Comment (jkarns275) [inline src/main/java/org/ngafid/WebServer.java:220]: This is also defined in your previous PR? Is your dark mode branch use this as its base?

File: webpack.config.js [hunk 3]
```diff
@@ -65,7 +66,7 @@ module.exports = {
         welcome: __dirname + "/src/main/javascript/welcome.js"
     },
 
-    devtool: "source-map",
+    devtool: false, //"source-map",
```
Comment (jkarns275) [inline webpack.config.js:69]: this should not be disabled i dont think?

**Review by AarC10 — CHANGES_REQUESTED** [review]

File: src/main/java/org/ngafid/SendEmail.java [hunk 4]
```diff
@@ -33,6 +32,16 @@ public class SendEmail {
 
     private static final Logger LOG = Logger.getLogger(SendEmail.class.getName());
 
+    private static final String baseURL = "https://ngafid.org";
+    //private static final String baseURL = "https://ngafidbeta.rit.edu";
```
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/SendEmail.java:36]: Yes

File: webpack.config.js (hunk 3 above)
Comment (jkarns275) [reply webpack.config.js:69]: This should be addressed

File: src/main/java/org/ngafid/SendEmail.java [hunk 5]
```diff
@@ -210,36 +293,74 @@ public static void sendEmail(ArrayList<String> toRecipients, ArrayList<String> b
...
+                    System.out.println("sending message (BCC)!");
+                    Transport.send(message);
+
+                }
 
-                // Send message
-                System.out.println("sending message!");
-                Transport.send(message);
-                System.out.println("Sent message successfully....");
+                System.out.println("Sent messages successfully....");
```
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/SendEmail.java:363]: real

File: src/main/java/org/ngafid/accounts/AirSyncFleet.java [hunk 6]
```diff
@@ -89,6 +89,59 @@ private AirSyncFleet(ResultSet resultSet) throws SQLException {
         }
     }
 
+    public boolean getOverride(Connection connection) throws SQLException {
+        String query = """
+            SELECT override from airsync_fleet_info WHERE
+                fleet_id = ?
+        """;
```
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/accounts/AirSyncFleet.java:96]: People are saying multi-line strings are the way to go

File: src/main/java/org/ngafid/accounts/AirSyncFleet.java [hunk 7]
```diff
@@ -89,6 +89,59 @@ private AirSyncFleet(ResultSet resultSet) throws SQLException {
...
+
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setInt(1, getId());
+
+            try (ResultSet rs = statement.executeQuery()) {
+                if (rs.next()) {
+                    return rs.getInt("override") != 0;
+                } else {
+                    return false;
+                }
```
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/accounts/AirSyncFleet.java:106]: I generally prefer an else statement to a drop down. Especially in a case like this where both branches are terminal.

File: src/main/java/org/ngafid/accounts/AirSyncFleet.java [hunk 8]
```diff
@@ -89,6 +89,59 @@ private AirSyncFleet(ResultSet resultSet) throws SQLException {
...
+        query.setInt(2, expected);
+
+        ResultSet rs = query.executeQuery();
+        if (rs.next()) {
+            rs.updateInt("mutex", newValue);
+            rs.updateRow();
+            return true;
+        } else {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/accounts/AirSyncFleet.java:142]: This code gets deleted in a later PR so can be ignored

File: src/main/java/org/ngafid/accounts/AirSyncFleet.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/accounts/AirSyncFleet.java:106]: Wait this is my code

File: src/main/java/org/ngafid/accounts/UserEmailPreferences.java [hunk 9]
```diff
@@ -25,8 +25,9 @@ public class UserEmailPreferences {
     private HashMap<String, Boolean> emailTypesUser;
     private String[] emailTypesKeys;
 
-    private static HashMap<Integer, User> users = new HashMap<Integer, User>();
-    private static HashMap<String, HashMap<String, Boolean>> emailTypesUsers = new HashMap<String, HashMap<String, Boolean>>();
+    private static HashMap<Integer, User> users = new HashMap<>();
+    private static HashMap<String, Integer> userIDs = new HashMap<>();
+    private static HashMap<String, HashMap<String, Boolean>> emailTypesUsers = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (jkarns275) [reply src/main/java/org/ngafid/accounts/UserEmailPreferences.java:30]: Should probably just update the getters to use the `Map` interface.

File: src/main/java/org/ngafid/accounts/EmailType.java [hunk 10]
```diff
@@ -75,10 +69,20 @@ public enum EmailType {
 
     //Store number of email types
     private final static int emailTypeCount = values().length;
+    private final static int emailTypeNonForcedCount;
```
Comment (jkarns275) [inline src/main/java/org/ngafid/accounts/EmailType.java:72]: `optionalEmailTypeCount`?

File: src/main/java/org/ngafid/accounts/EmailType.java [hunk 11]
```diff
@@ -75,10 +69,20 @@ public enum EmailType {
...
 
     private static Logger LOG = Logger.getLogger(EmailType.class.getName());
 
     static {
+
```
Comment (jkarns275) [inline src/main/java/org/ngafid/accounts/EmailType.java:77]: comment

File: src/main/java/org/ngafid/accounts/UserEmailPreferences.java [hunk 12]
```diff
@@ -48,7 +49,14 @@ public UserEmailPreferences(int userId, HashMap<String, Boolean> emailTypesUser)
 
     public static void addUser(Connection connection, User user) throws SQLException {
 
-        users.put(user.getId(), user);
+        int userID = user.getId();
```
Comment (jkarns275) [inline src/main/java/org/ngafid/accounts/UserEmailPreferences.java:52]: Add a comment just to say this is updating the user cache maps

**Comment by jkarns275** [comment]
honestly LGTM fix conflicts and we send it

**Review by AarC10 — APPROVED** [review]

**Pull request closed without merging**